libsql-server/src/namespace/mod.rs

```diff
@@ -360,9 +361,24 @@ struct NamespaceStoreInner<M: MakeNamespace> {
 
 impl<M: MakeNamespace> NamespaceStore<M> {
     pub fn new(make_namespace: M, allow_lazy_creation: bool, snapshot_at_shutdown: bool) -> Self {
+        let store = Cache::<NamespaceName, NamespaceEntry<M::Database>>::builder()
+            .async_eviction_listener(|name, ns, _| {
+                Box::pin(async move {
+                    tracing::info!("namespace `{name}` deallocated");
+                    // shutdown namespace
+                    if let Some(ns) = ns.write().await.take() {
+                        if let Err(e) = ns.destroy().await {
+                            tracing::error!("error deallocating `{name}`: {e}")
+                        }
+                    }
+                })
+            })
+            // TODO(marin): configurable capacity
+            .max_capacity(25)
```

Comment:
@MarinPostma does it imply we cap our active databases at 25? Then we need to definitely make it configurable, but also bump it to at least a 100, even for those small 256MiB machines.

Reply:
And https://docs.rs/moka/latest/moka/future/struct.CacheBuilder.html#method.time_to_idle `time_to_idle` sounds very relevant for us too, accessing the db bumps the counter, but it gets automatically evicted after some time (and be generous with that time, like, minutes)

Reply:
yeah, it was still very POC and I hit other issues even before I started tuning the eviction logic